### ISSUE TITLE
Behavior i18n & versionnable

### ISSUE THREAD

**Issue by loicdev** [propelorm/Propel2#880]
Hi ,
can i  put  i18n field using by my i18n behavior in the versionning table (versionnable) ?

I try it but only no translate field are versionning in {table}_version .

Anyone have the same issue as me ?

**Comment by marcj** [comment]
The order is important here. i18n needs to be placed first in your schema.xml, then the versionable.

**Comment by loicdev** [comment]
First thanks for answer.
It works partly because i18n fields are still not versionnated..
In my apps , i want to used at least four behaviors ( Versionnable , nested_set, i18n ,sluggable, archivable ).

Have you got a concrete and simple example for articles and category entity?

**Comment by marcj** [comment]
You could provide us a fiddle at http://sandbox.propelorm.org/ so we can reproduce you issue easier and faster. Maybe it's really only a behavior order issue.

**Comment by loicdev** [comment]
Here is my sample : http://sandbox.propelorm.org/2cff3d7
As i have said i want to the field name,url and locale go to my node_version table.

Also i have some problems with inheritance on Framework Symfony 2 in Class inheritance Mode.

I can't initialize my SiteType Form with url,name field which are in my NodeType Form.

**Comment by apinnecke** [comment]
@loicdev There is no `locale` field in your `node` table, so why do you expect it to appear in your `node_version` table? It's only available in `node_i18n` table, so from my point of view the only way to have it in `node_version` is to add it to `node` and set it manually in `preSave` hook. Or am i missing something?

**Comment by ck86** [comment]
@apinnecke The `locale` field will be created by the i18n behavior and is not part of the `node` table, but of `node_i18n`, right? 

I think the i18n behavior should be also versionable if the schema is versionable.

**Comment by apinnecke** [comment]
@ck86 Yep, alright, but then it should probably have an `node_i18n_version` table.

**Comment by ck86** [comment]
And it should all be done in one table definition ;)

**Comment by rswebdev** [comment]
This is just to vote for a fix.

**Comment by motin** [comment]
I believe the proper way to fix this is to make sure that the _i18n-relation is versioned. However, as pointed out on http://stackoverflow.com/a/17807299/682317 and http://propelorm.org/documentation/behaviors/versionable.html, this is not working at the moment:
> Versioning of related objects is only possible for simple foreign keys. Relationships based on composite foreign keys cannot preserve relation versionning for now.

Since i18n adds complex foreign keys, the two won't work together.
